Generate the mousePhenotype dataset

tskir opened the pull request:
See: https://github.com/opentargets/platform/issues/1471, https://github.com/opentargets/platform/issues/1642.

* Thoroughly reorganise preparation and joining for all data types
* Generate and output the `mousePhenotypes` dataset
* Fetch, parse and process MP high level classes
* Add two new dependencies (`retry` and `pronto`) and update the execution environments

DSuveges commented on modules/PhenoDigm.py:
```diff
@@ -234,53 +249,104 @@ def load_data_from_cache(self):
         )
+        # Process MP definitions to extract high level classes for each term.
+        mp = pronto.Ontology(os.path.join(self.cache_dir, self.MGI_MP_FILENAME))
+        mp_class = [
```
OK, what happened is the default pronto version was 0.1.2 instead 2.4.3 specified in the env file.

DSuveges commented on modules/PhenoDigm.py:
```diff
@@ -382,23 +418,49 @@ def generate_phenodigm_evidence_strings(self, score_cutoff):
+            logging.info(f'Processing dataset {outfile}')
+            with tempfile.TemporaryDirectory() as tmp_dir_name:
+                (
+                    dataset.coalesce(1).write.format('json').mode('overwrite')
```
I think considering the file size coalescing is a risky step. Even the score threshold is set to 80, the number of phenotypes is 550k+. When using the default cutoff (0) even a very memory rich VM failed.

tskir replied:
Hmm, just out of curiosity, how much memory were you using? It's just that it seems to work on my desktop (64 GB RAM) even without any threshold. Although I haven't measured how much memory is actually used.

Also, I think we need to revisit this question in general (can discuss during the stand-up). This needs to be uniform across all parsers: they need to output either a single JSON file (this could be done without coalescing by just concatenating all JSON chunks together), or to always output a directory with JSONs.

tskir commented on modules/PhenoDigm.py:
```diff
@@ -234,53 +249,104 @@ def load_data_from_cache(self):
+            )
+        )
+
+    def generate_phenodigm_evidence_strings(self, score_cutoff):
```
Yes, currently this is the case. I believe @ireneisdoomed had some thoughts & back of the envelope calculations for some sensible threshold values which we could pick, so that the results are broadly comparable to the previous ones.

tskir commented:
@DSuveges @ireneisdoomed Additional updates as per discussions:
* Removed root element from high level phenotype classes
* Added partial schema nesting
* Updated environments
* Rebased against master
* Addressed the comments
* Ran and generated the data

Change in the total number of phenotypes:
* Before 557010
* After 189971

Here are the actual phenotypes: [phenodigm-phenotypes-2021-09-02.json.gz](https://github.com/opentargets/evidence_datasource_parsers/files/7097004/phenodigm-phenotypes-2021-09-02.json.gz)

They now look something like this:
![image](https://user-images.githubusercontent.com/10669118/131803138-01da8c4b-5868-41b3-84cb-209ba1c65fc0.png)

Ready for the next review iteration.
